service/src/main/java/org/dice_research/fc/paths/verbalizer/MultiplePathVerbalizer.java

```diff
@@ -0,0 +1,641 @@
+package org.dice_research.fc.paths.verbalizer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.aksw.jena_sparql_api.core.QueryExecutionFactory;
+import org.aksw.triple2nl.TripleConverter;
+import org.apache.commons.math3.util.Pair;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.Resource;
+import org.dice_research.fc.data.IPieceOfEvidence;
+import org.dice_research.fc.data.QRestrictedPath;
+import org.dice_research.fc.util.RDFUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.collect.Lists;
+import simplenlg.features.Feature;
+import simplenlg.framework.CoordinatedPhraseElement;
+import simplenlg.framework.DocumentElement;
+import simplenlg.framework.NLGElement;
+import simplenlg.framework.NLGFactory;
+import simplenlg.framework.WordElement;
+import simplenlg.lexicon.Lexicon;
+import simplenlg.phrasespec.SPhraseSpec;
+import simplenlg.realiser.english.Realiser;
+
+/**
+ * Verbalizer that supports the verbalization of paths of length up to 3. Paths of length 3 are
+ * decomposed into paths of length 2 and verbalized as 2+1 instead.
+ * 
+ * @author Alexandra Silva
+ *
+ */
+public class MultiplePathVerbalizer extends DefaultPathVerbalizer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MultiplePathVerbalizer.class);
+
+  private static final String NOT_FOUND =
+      "Verbalization not found as no intermediate nodes were found for the meta-path.";
```

Comment:
To enable a better error handling, it would be good if in case of such an error you either return `null` or some other value that can be identified easily, or you throw an exception.
At the moment, it will be very hard for a user to figure out that the string they receive is not a verbalization but an error message.